Firmware/mmu2_protocol_logic.cpp

```diff
@@ -818,6 +818,11 @@ StepStatus ProtocolLogic::Step() {
                 currentStatus = Processing;
             }
         }
+
+        if (ExpectsResponse()) {
```

Comment:
Is it really relevant at this level of code? Shouldn't it be in the `else` block?

IMO if we activated a planned request (which is the case of ReadRegister), it should step into the `else` block ... may be the condition:
```
if (!previousCommandFinished && currentScope == Scope::Command) {
```
should be changed to
```
if(ExpectsResponse()){
```
What do you think?

Reply:
@gudnimg Is this solved?

Reply:
We cant use the else block because the if statement will always be true in this scenario: `if (!ActivatePlannedRequest())`, where there is no planned request waiting but we are expecting a response from a previously “activated planned request”.

The bug here may be a planned request is always activated regardless if we’ve not received the response yet. For example if 3 requests were theoretically planned, they’d all be activated after the other and the responses are “ignored”. 

I didnt touch the if-else since I was worried about breaking things, but I suspect it is wrong.

Reply:
@DRracer I've got the unit tests working on the MK4 side now. I tried your suggestion where

```c
if (!previousCommandFinished && currentScope == Scope::Command) {
```

is changed to:

```c
if(ExpectsResponse()){
```

And it also passes the unit test! 👍 I'll test this on the hardware now, and if it works then I'll submit the changes. If it doesn't work, then I guess I'll need to update the unit tests ;)

Reply:
I can confirm the fix works.
![image](https://github.com/prusa3d/Prusa-Firmware/assets/8218499/5285b7fd-0b06-482e-9f72-ffacb7703f4c)

Reply:
Here is an example where the fix is working for register writes. The "Expecting response!" log is in the same line as we have `currentStatus = Processing;`
![image](https://github.com/prusa3d/Prusa-Firmware/assets/8218499/3c8684d3-b7f4-4b65-b03e-cb54954d5df4)